app/models/manageiq/providers/redhat/infra_manager/event_fetcher.rb

```diff
@@ -0,0 +1,486 @@
+class ManageIQ::Providers::Redhat::InfraManager::EventFetcher
+  attr_reader :ext_management_system
+
+  def initialize(ems)
+    @ext_management_system = ems
+  end
+
+  def events(query_options)
+    query = {}
+    query[:max] = query_options[:max] if query_options[:max]
+    query[:from] = query_options[:since] if query_options[:since]
+    ext_management_system.with_provider_connection(:version => 4) do |ems_service|
+      events = ems_service.system_service.events_service.list(query)
+      events.each { |event| set_event_name!(event) }
+      events
+    end
+  end
+
+  def set_event_name!(event)
+    name = EVENT_CODES[event.code] || "UNKNOWN"
+    event.name = name
+  end
+
+  EVENT_CODES = {
```

Comment:
Have you checked the correctness with [1]? I remember that there were some inconsistencies but not sure whether we fixed those or not.

[1] https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/AuditLogType.java